### ISSUE TITLE
Fine-tune with Matting data.

### ISSUE THREAD

**Issue by drew0523** [ZhengPeng7/BiRefNet#223]
Hi, I really appreciate your work.

I'm trying to fine-tune BiRefNet using my custom matting dataset. Since BiRefNet is originally designed for segmentation tasks, do I need to modify its architecture for matting?

I noticed that there's a pre-trained weight file named BiRefNet-matting-epoch_100.pth, but I'm having trouble understanding how matting works within the standard BiRefNet architecture.

**Comment by ZhengPeng7** [comment]
What problem do you expect in your mind? You can even use vanilla U-Net to do both seg and mat, only if the loss is different. But of course, the sigmoid at the head is not part of the model.

**Comment by drew0523** [comment]
Thanks I understand. But I have additional question.

def inference(model, data_loader_test, pred_root, method, testset, device=0):
    model_training = model.training
    if model_training:
        model.eval()
    for batch in tqdm(data_loader_test, total=len(data_loader_test)) if 1 or config.verbose_eval else data_loader_test:
        inputs = batch[0].to(device)
        label_paths = batch[-1]
        with autocast_ctx, torch.no_grad():
            ### scaled_preds = model(inputs)[-1].sigmoid().to(torch.float32)

In inference.py, the last line of the code returns an output in the range of 0 to 1. How is that suitable for Dichotomous Image Segmentation? Shouldn't it return a binary output instead? According to this code, it seems to produce continuous values between 0 and 1.

**Comment by ZhengPeng7** [comment]
If you take a look into similar tasks, such as SOD, COD, and med seg, you'll find that these tasks all use sigmoid at the end to control the output in range of (0, 1). Ideally, they're 0 or 1. But practically, they're 0~1. I also used some other techniques in my previous works like [GCoNet+](https://github.com/ZhengPeng7/GCoNet_plus) to make the prediction more thresholded to 0 or 1. If you're interested, you can also search and think about more skills about it.

**Comment by drew0523** [comment]
Thank you so  much! I really appreciate for your work!
